File: src/main/java/com/medicare/backend/models/Doctor.java
package com.medicare.backend.models;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "doctors")
public class Doctor {
    @Id
    private String id;

    @Indexed
    private String userId;

    private String name;

    @Indexed
    private String specialty;

    private String experience;

    private Double rating = 0.0;

    private String imageUrl;

    private List<String> availableSlots;

    private String status = "online"; // online, offline

    @CreatedDate
    private LocalDateTime createdAt;

    @LastModifiedDate
    private LocalDateTime updatedAt;
}

File: src/models/Doctor.ts
import mongoose, { Schema, Model } from 'mongoose';
import { IDoctor } from '../types';

const doctorSchema = new Schema<IDoctor>(
    {
        userId: {
            type: String,
            required: true,
            ref: 'User',
        },
        name: {
            type: String,
            required: true,
        },
        specialty: {
            type: String,
            required: true,
        },
        experience: {
            type: String,
            required: true,
        },
        rating: {
            type: Number,
            default: 0,
            min: 0,
            max: 5,
        },
        imageUrl: {
            type: String,
            default: '',
        },
        availableSlots: {
            type: [String],
            default: [],
        },
        status: {
            type: String,
            enum: ['online', 'offline'],
            default: 'online',
        },
    },
    {
        timestamps: true,
    }
);

// Create index
doctorSchema.index({ userId: 1 });
doctorSchema.index({ specialty: 1 });

export const Doctor: Model<IDoctor> = mongoose.model<IDoctor>('Doctor', doctorSchema);
